file/builder.go

```diff
@@ -766,11 +784,47 @@ func (b *stateBuilder) ingestRoute(r FRoute) error {
 	var plugins []FPlugin
 	for _, p := range r.Plugins {
 		p.Route = &kong.Route{ID: kong.String(*r.ID)}
+		if err := b.ingestPluginDefaults(p); err != nil {
+			return err
+		}
 		plugins = append(plugins, *p)
 	}
 	return b.ingestPlugins(plugins)
 }
 
+func (b *stateBuilder) getPluginSchema(entityID *string) (map[string]interface{}, error) {
+	var schema map[string]interface{}
+
+	// lookup in cache
+	if schema, ok := b.schemas[*entityID]; ok {
+		return schema, nil
+	}
+	schema, err := b.client.Plugins.GetFullSchema(b.ctx, entityID)
+	if err != nil {
+		return schema, err
+	}
+	b.schemas[*entityID] = schema
+	return schema, nil
+}
+
+func (b *stateBuilder) ingestPluginDefaults(plugin *FPlugin) error {
+	if b.client == nil {
+		return nil
+	}
+	schema, err := b.getPluginSchema(plugin.Name)
+	if err != nil {
+		return err
```

Comment:
```suggestion
		return fmt.Errorf("retrieve schema for "%v" from Kong: %v", *plugin.Name, err)
```

Annotate error where possible. You can skip this in lower-level library code but as the code gets "closer" to the user, do more and more of it. It contextualizes the error and makes it a bit more actionable for the user.

Reply:
done